lib/collection/src/update_handler.rs

```diff
@@ -285,18 +335,27 @@ impl UpdateHandler {
         optimization_handles: Arc<TokioMutex<Vec<StoppableTaskHandle<bool>>>>,
         max_handles: usize,
     ) {
-        while let Some(signal) = receiver.recv().await {
-            match signal {
-                OptimizerSignal::Nop | OptimizerSignal::Operation(_) => {
+        loop {
+            let receiver = timeout(OPTIMIZER_CLEANUP_INTERVAL, receiver.recv());
+            let result = receiver.await;
+
+            // Always clean up on any signal
+            Self::cleanup_optimization_handles(optimization_handles.clone()).await;
+
+            match result {
+                // Channel closed or stop signal
+                Ok(None | Some(OptimizerSignal::Stop)) => break,
+                // Clean up interval
+                Err(_) => continue,
```

Comment:
What about using the full type like `Err(Elapsed)`?
It would make it clearer to the reader which error we are handling.
It might also prevent some catch all errors bug in a future refactoring.

Reply:
Good idea. Updating.